File: logicaDeClases/java/Dinodado.java
package logicaDeClases.java;

import java.util.Random;

public class Dinodado 
{
    // DEFINE LAS CARAS DEL DADO
    public enum DiceType {
        BOSQUE,
        ROCOSA,
        ASEOS,
        NOREX,
        NODINO,
        CAFE
    }

    private DiceType currentFace;
    private static final Random random = new Random();

    // CONSTRUCTOR DEL DADO 
    public Dinodado() {
        roll();
    }

    // LANZADA DE DADO
    public void roll() {
        DiceType[] faces = DiceType.values();
        int index = random.nextInt(faces.length);
        currentFace = faces[index];
    }

    // OBTENER LA CARA RESULTANTE DESPUES DEL ROLL DEL DADO
    public DiceType getCurrentFace() {
        return currentFace;
    }

    // METODO PARA MOSTRAR LA CARA DEL DADO
    public String toString() {
        return "Dice showing: " + currentFace.name();
    }

    // METODO PRINCIPAL PARA PROBAR EL DADO
    public static void main(String[] args) {
        Dinodado dice = new Dinodado();
        dice.roll();
        System.out.println("Cara resultante: " + dice.getCurrentFace());
    }
}

File: draftosaurus-project/app/models/Dinodado.php
<?php

    class Dinodado 
    {
        // DEFINE LAS CARAS DEL DADO
        private const DICE_TYPES = [
            'BOSQUE',
            'ROCOSA',
            'ASEOS',
            'NOREX',
            'NODINO',
            'CAFE'
        ];

        private string $currentFace;

        // CONSTRUCTOR DEL DADO
        public function __construct() {
            $this->roll();
        }

        // LANZADA DE DADO
        public function roll(): void {
            $index = random_int(0, count(self::DICE_TYPES) - 1);
            $this->currentFace = self::DICE_TYPES[$index];
        }

        // OBTENER LA CARA RESULTANTE DESPUES DEL ROLL DEL DADO
        public function getCurrentFace(): string {
            return $this->currentFace;
        }

        // METODO PARA MOSTRAR LA CARA DEL DADO
        public function __toString(): string {
            return "Cara resultante: {$this->currentFace}";
        }
    }

    // METODO PRINCIPAL PARA PROBAR EL DADO
    if (basename(__FILE__) == basename($_SERVER['PHP_SELF'])) {
        $dice = new Dinodado();
        $dice->roll();
        echo "Cara resultante: " . $dice->getCurrentFace() . "\n";
    }